### PR TITLE
enhance: limit getSegmentInfo batch size to avoid excced grpc message limit

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #35395

### CODE REVIEW

File: internal/datanode/channel/channel_manager.go [hunk 1]
```diff
@@ -410,6 +410,7 @@ func (r *opRunner) watchWithTimer(info *datapb.ChannelWatchInfo) *opState {
 		defer finishWaiter.Done()
 		fg, err := r.watchFunc(ctx, r.pipelineParams, info, tickler)
 		if err != nil {
+			log.Warn("Fail to watch", zap.String("channel", r.channel), zap.Error(err))
```
Comment (XuanYang-cn) [inline internal/datanode/channel/channel_manager.go:413]: channel is already logged with in Line:352

File: internal/flushcommon/broker/datacoord.go [hunk 2]
```diff
@@ -62,30 +64,47 @@ func (dc *dataCoordBroker) ReportTimeTick(ctx context.Context, msgs []*msgpb.Dat
...
+		}
+		return infoResp, nil
 	}
-	err = binlog.DecompressMultiBinLogs(infoResp.GetInfos())
-	if err != nil {
-		log.Warn("Fail to DecompressMultiBinLogs", zap.Error(err))
-		return nil, err
+
+	ret := make([]*datapb.SegmentInfo, 0, len(ids))
+	batchSize := 100
```
Comment (xiaofan-luan) [inline internal/flushcommon/broker/datacoord.go:95]: batch can be 1000?

**Review by XuanYang-cn — APPROVED** [review]
/lgtm

**Pull request merged**